tsconfig.build.json

```diff
@@ -1,23 +0,0 @@
-{
-  "extends": "./tsconfig.json",
-  "compilerOptions": {
-    "declaration": true,
-    "declarationMap": true,
```

Comment:
I see that we don't need this file anymore because `tsup` uses an entry point. That's cool, however, it appears that [declaration maps](https://www.typescriptlang.org/tsconfig#declarationMap) aren't being generated anymore. These files make it possible to jump directly from the usage of a thing in a library to the definition of that thing (and not just its type definition, which is the default).

I see in the `tsup` docs that they [recommend excluding declaration map files from published packages](https://tsup.egoist.dev/#generate-typescript-declaration-maps--d-ts-map); they seem to indicate that they're only useful when navigating a monorepo. I disagree with this viewpoint, however, and it seems like [others do, too](https://github.com/egoist/tsup/issues/885). The other thing is that the package they use to generate declaration files, `rollup-plugin-dts`, [doesn't support generating declaration maps](https://github.com/egoist/tsup/issues/564#issuecomment-1032182885). Someone has suggested [replacing this library with `@rollup/plugin-typescript`](https://github.com/egoist/tsup/issues/885#issuecomment-1685110836). Note that this may require bringing back this file (or somehow instructing TypeScript with `declarationMap: true` without defining an explicit tsconfig).

Obviously there's a lot going on here, so have you considered this and what do you think about this solution?

Reply:
I didn't include them because of the documentation mentioning that they shouldn't be included in an NPM package, but you're right that they are actually useful to include.

In that case, I think we should remove declaration generation from the `tsup` config, and just let TypeScript handle the declarations and declaration map. The only "problem" is that TypeScript will generate a declaration file for each included file, which doesn't match the build output by `tsup` with outputs just a single file for each bundle type. Our `package.json` doesn't allow importing anything other than the index anyway, so I don't think it will be a problem in practice.

Reply:
I restored the `tsconfig.build.json` file, and changed it so we now use `tsc` for generating the declarations and declaration maps. The folder structure of `dist` now looks like:

```
dist/
├─ types/
│  ├─ index.d.ts
│  ├─ index.d.ts.map
│  ├─ file-a.d.ts
│  ├─ file-a.d.ts.map
│  ├─ etc.
├─ index.js
├─ index.js.map
├─ index.mjs
├─ index.mjs.map
```